libs/processors/src/main/java/org/wordpress/android/processor/RemoteConfigProcessor.kt

```diff
@@ -1,128 +1,128 @@
-            generateRemoteFeatureConfigCheck(remoteFeatureNames)
-            generateFeaturesInDevelopment(featuresInDevelopment)
-            true
-        } else {
-            false
+
+@OptIn(KspExperimental::class)
+class RemoteConfigProcessor(
+    private val codeGenerator: CodeGenerator,
+) : SymbolProcessor {
+    /**
+     * See: https://github.com/google/ksp/issues/797#issuecomment-1041127747
+     * Also: https://github.com/google/ksp/blob/a0cd7774a7f65cec45a50ecc8960ef5e4d47fc21/examples/playground/test-processor/src/main/kotlin/TestProcessor.kt#L20
+     */
+    private var invoked = false
+
+    override fun process(resolver: Resolver): List<KSAnnotated> {
+        if (invoked) {
+            return emptyList()
         }
+
+        val remoteFeatures = resolver.getSymbolsWithAnnotation("org.wordpress.android.annotation.Feature")
+            .toList()
+
+        generateRemoteFeatureConfigDefaults(resolver, remoteFeatures)
+        generateRemoteFieldsConfigDefaults(resolver)
+        generateFeaturesInDevelopment(resolver)
+        generateRemoteFeatureConfigCheck(remoteFeatures)
+
+        invoked = true
+        return emptyList()
     }
 
-    @Suppress("TooGenericExceptionCaught", "SwallowedException")
-    private fun generateRemoteFeatureConfigDefaults(
-        remoteConfigDefaults: Map<String, String>
-    ) {
-        try {
-            val fileContent = RemoteFeatureConfigDefaultsBuilder(remoteConfigDefaults).getContent()
-
-            val kaptKotlinGeneratedDir = processingEnv.options[KAPT_KOTLIN_GENERATED_OPTION_NAME]
-            fileContent.writeTo(File(kaptKotlinGeneratedDir))
-        } catch (e: Exception) {
-            processingEnv.messager.printMessage(Kind.ERROR, "Failed to generate remote feature config defaults")
+    private fun generateRemoteFeatureConfigDefaults(resolver: Resolver, remoteFeatures: List<KSAnnotated>) {
+        val experiments = resolver.getSymbolsWithAnnotation("org.wordpress.android.annotation.Experiment")
+            .toList()
+
+        val defaults = (remoteFeatures + experiments)
+            .map { element: KSAnnotated ->
+                val featuresDefaults = element.getAnnotationsByType(Feature::class)
+                    .toList().associate { annotation ->
+                        annotation.remoteField to annotation.defaultValue.toString()
+                    }
+                val experimentsDefaults = element.getAnnotationsByType(Experiment::class).toList()
+                    .toList().associate { annotation ->
+                        annotation.remoteField to annotation.defaultVariant
+                    }
+                featuresDefaults + experimentsDefaults
+            }.flatMap { it.toList() }
+            .toMap()
+
+        if (defaults.isNotEmpty()) {
+            RemoteFeatureConfigDefaultsBuilder(defaults).getContent()
+                .writeTo(
+                    codeGenerator,
+                    aggregating = true,
```

Comment:
All output files are set to `aggregating` strategy for incremental processing. They have to, as we gather annotations from all available files in the codebase. 

See https://kotlinlang.org/docs/ksp-incremental.html#aggregating-vs-isolating for more details.